src/core/python2js.c

```diff
@@ -190,89 +189,128 @@ _python2js_bytes(PyObject* x)
   return hiwire_bytes(x_buff, length);
 }
 
+/** WARNING: This function is not suitable for fallbacks. If this function
+ * returns NULL, we must assume that the cache has been corrupted and bail out.
+ */
```

Comment:
So the cache is corrupted because we are never calling `_python2js_remove_from_cache` on failure unlike before, as far as I understand?

Reply:
That's correct, but the code was wrong before too. The minimal counterexample is a bit complicated though. Let me cook one up.

Reply:
Actually now I think it was actually right before too. Confusing.

Reply:
But yeah you understand correctly.